iphone/Classes/GeolocationModule.m

```diff
@@ -1093,6 +1117,15 @@ - (void)locationManager:(CLLocationManager *)manager didChangeAuthorizationStatu
                 code = 1;
                 errorStr = @"The user denied access to use location services.";
         }
+      
+        // This is very important for iOS 11+ because even if the user did an incremental authorization before
+        // (by selecting "when in use", he/she will still get a dialog that includes the "when in use" option.
+        // In case that one is still selected then, the developer should know about that selection and the following
+        // statement allows that.
+        if (!requestedStatusMatchesActualStatus) {
```

Comment:
@hansemannn this addition has zero effect, because when the authorization status is unchanged and thus this whole method is not fired.

Reply:
It's for the initial "when in use" instead of "always" selection, which would fire but with `success: true`.

Reply:
You're right